File: brn.jl
#!/usr/bin/env julia

bd = pwd()
dirs = readdir(bd)
for dir in dirs
  files = readdir(dir)
  for file in files
    fop = abspath(dir) * "/" * file
    fnp = abspath(bd) * "/" * dir * " - " * file
    mv(fop, fnp)
  end
end

File: brn.go
package main

import (
	"fmt"
	"os"
	"io/ioutil"
)
	
func main() {
	bd, _ := os.Getwd()
	dirs, _ := ioutil.ReadDir(bd)
	for _, dir := range dirs {
		prnt := dir.Name()
		files, _ := ioutil.ReadDir(prnt)
		for _, file := range files {
			oname := file.Name()
			nname := prnt + " - " + oname
			fop := fmt.Sprintf("%s/%s/%s", bd, prnt, oname)
			fnp := fmt.Sprintf("%s/%s", bd, nname)
			os.Rename(fop, fnp)
		}
	}
}
